rxandroid/src/test/java/io/reactivex/android/plugins/RxAndroidPluginsTest.java

```diff
@@ -67,32 +72,99 @@ public void resetClearsMainThreadHandler() {
-                schedulerRef.set(scheduler);
-                return newScheduler;
-            }
-        });
+        RxAndroidPlugins
+                .setInitMainThreadSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
+                    @Override public Scheduler apply(Callable<Scheduler> scheduler) {
+                        schedulerRef.set(scheduler);
+                        return newScheduler;
+                    }
+                });
 
-        Scheduler scheduler = new EmptyScheduler();
+        Callable<Scheduler> scheduler = new Callable<Scheduler>() {
+            @Override public Scheduler call() throws Exception {
+                return new EmptyScheduler();
+            }
+        };
         Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(scheduler);
         assertSame(newScheduler, actual);
         assertSame(scheduler, schedulerRef.get());
     }
 
     @Test
-    public void resetClearsInitMainThreadHandler() {
-        RxAndroidPlugins.setInitMainThreadSchedulerHandler(new Function<Scheduler, Scheduler>() {
-            @Override public Scheduler apply(Scheduler scheduler) {
-                throw new RuntimeException();
+    public void resetClearsInitMainThreadHandler() throws Exception {
+        RxAndroidPlugins
+                .setInitMainThreadSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
+                    @Override public Scheduler apply(Callable<Scheduler> scheduler) {
+                        throw new RuntimeException();
+                    }
+                });
+
+        final Scheduler scheduler = new EmptyScheduler();
+        Callable<Scheduler> schedulerCallable = new Callable<Scheduler>() {
+            @Override public Scheduler call() throws Exception {
+                return scheduler;
             }
-        });
+        };
+
         RxAndroidPlugins.reset();
 
-        Scheduler scheduler = new EmptyScheduler();
-        Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(scheduler);
-        assertSame(scheduler, actual);
+        Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(schedulerCallable);
+        assertSame(schedulerCallable.call(), actual);
     }
+
+    @Test
+    public void defaultMainThreadSchedulerIsInitializedLazily() {
+        final Function<Callable<Scheduler>, Scheduler> safeOverride = new Function<Callable<Scheduler>, Scheduler>() {
+            @Override public Scheduler apply(Callable<Scheduler> __) {
+                return new EmptyScheduler();
+            }
+        };
+        final Callable<Scheduler> unsafeDefault = new Callable<Scheduler>() {
+            @Override public Scheduler call() throws Exception {
+                throw new AssertionError(
+                        "Default Scheduler instance should not have been evaluated");
+            }
+        };
+
+        // unsafe default Scheduler Callable should not be evaluated when overriding
+        try {
```

Comment:
This whole try/catch isn't needed. The plugins are reset before and after each test.